src/rust/engine/graph/src/lib.rs

```diff
@@ -287,12 +291,18 @@ impl<N: Node> InnerGraph<N> {
   /// The Walk will iterate over all nodes that descend from the roots in the direction of
```

Comment:
Probably worth updating the do comment to add: "until meeting a node that matches the `stop_walking_predicate`"